changelog/fragments/01-k8s-kubebuilder-bump.yaml

```diff
@@ -0,0 +1,34 @@
+# entries is a list of entries to include in
+# release notes and/or the migration guide
+entries:
+  - description: >
+      Bump supported Kubernetes version to 1.27
+    kind: "change"
+    breaking: false
+  - description: >
+      (helm/v1): Deprecate the `--config` flag due to the controller-runtime ComponentConfig
+      package being deprecated with the planned removal of support for loading manager options
+      from a file
+    kind: "deprecation" 
+    breaking: false
+  - description: >
+      (go/v4): go/v4 is now stable and is the default version
+      used when scaffolding a Go based operator
+    kind: "change"
+    breaking: false
+  - description: >
+      (kustomize/v2): kustomize/v2 is now stable and is the default version
+      used in the plugin chain when using go/v4, ansible/v1, helm/v1, and hybrid/v1-alpha plugins
+    kind: "change"
```

Comment:
>Needs changelog.
Do you mean migration? If so, why? This only impacts net new operators as far as I am aware.

I can mark these as breaking changes

Reply:
The reason why I'm pushing back on adding migration for this is because that is a _large_ migration effort. Writing up the migration doc for this would be a significant scope creep for this PR IMO. If it is absolutely necessary to have the migration guide I'd prefer if we could separate that into a follow up PR

Reply:
> Do you mean migration? If so, why? 

Sorry, yes, I meant migration. 

> This only impacts net new operators as far as I am aware.

Not necessarily. When an existing operator bumps Operator SDK, its always recommended to also bump the plugin version to ensure that the latest changes are supported. For example, if there is any bug or feature that users want in go/v2, we would not support it. Which is why, bumping plugin is equally important for existing users. In order to do so, we need to explicitly mention the changes which have been made in the newer plugin versions.

Reply:
Thanks for the reference, I didn't realize there was already a kubebuilder migration guide. Will update the changelog

Reply:
https://github.com/operator-framework/operator-sdk/pull/6613/commits/b5e42f8519268da754d4d80bcc3225b758433eb6